### ISSUE TITLE
Add support for Auth

### ISSUE THREAD

**Issue by davissp14** [kragniz/python-etcd3#98]
Doesn't look like there's anyway to provide user credentials to the etcd.client object.

**Comment by kragniz** [comment]
Yeah, I'd like this to be added, it's the last big chunk of API coverage we're missing (see https://github.com/kragniz/python-etcd3/issues/50).

**Comment by davissp14** [comment]
Looking at how to support this for both Python and Ruby, it seems unclear how we would actually use the  generated Authentication token.  Have you done much digging on this yet?

**Comment by joe-pll** [comment]
Hi, with my company we required to have authentication and users/roles methods. We implemented those, but we don't have time to write all the documentation and the tests. If you want to have look at it and maybe change what you need then I'll make the PR. Let me know.
